### PR TITLE
Add interpolation target for user-specified points

### PR DESCRIPTION
**Pull request by fmahebert**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/NumericalAlgorithms/Interpolation/InterpolationTargetSpecifiedPoints.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+namespace intrp::OptionHolders {
+
+template <size_t VolumeDim>
+SpecifiedPoints<VolumeDim>::SpecifiedPoints(
+    std::vector<std::array<double, VolumeDim>> points_in) noexcept
+    : points(std::move(points_in)) {}
+
+template <size_t VolumeDim>
+void SpecifiedPoints<VolumeDim>::pup(PUP::er& p) noexcept {
+  p | points;
```
Comment (kidder) [inline src/NumericalAlgorithms/Interpolation/InterpolationTargetSpecifiedPoints.cpp:20]: include pup

File: src/NumericalAlgorithms/Interpolation/InterpolationTargetSpecifiedPoints.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#include "NumericalAlgorithms/Interpolation/InterpolationTargetSpecifiedPoints.hpp"
+
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace intrp::OptionHolders {
+
+template <size_t VolumeDim>
+SpecifiedPoints<VolumeDim>::SpecifiedPoints(
+    std::vector<std::array<double, VolumeDim>> points_in) noexcept
+    : points(std::move(points_in)) {}
```
Comment (kidder) [inline src/NumericalAlgorithms/Interpolation/InterpolationTargetSpecifiedPoints.cpp:16]: include utility

File: src/NumericalAlgorithms/Interpolation/InterpolationTargetSpecifiedPoints.hpp [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "NumericalAlgorithms/Interpolation/Tags.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
```
Comment (kidder) [inline src/NumericalAlgorithms/Interpolation/InterpolationTargetSpecifiedPoints.hpp]: unused

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash

**Comment by fmahebert** [comment]
I've updated the includes. Thanks for catching these @kidder

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**